test/distrib/cpp/run_distrib_test_raspberry_pi.sh

```diff
@@ -0,0 +1,90 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -ex
+
+cd "$(dirname "$0")/../../.."
+grpc_dir=$(pwd)
+
+echo "deb http://archive.debian.org/debian jessie-backports main" | tee /etc/apt/sources.list.d/jessie-backports.list
+echo 'Acquire::Check-Valid-Until "false";' > /etc/apt/apt.conf
+sed -i '/deb http:\/\/deb.debian.org\/debian jessie-updates main/d' /etc/apt/sources.list
+apt-get update
+apt-get install -t jessie-backports -y git libssl-dev wget
+
+# Install CMake 3.16
+wget -q -O cmake-linux.sh https://github.com/Kitware/CMake/releases/download/v3.16.1/cmake-3.16.1-Linux-x86_64.sh
+sh cmake-linux.sh -- --skip-license --prefix=/usr
+rm cmake-linux.sh
+
+# Build protoc and grpc_cpp_plugin for the host architecture.
+mkdir -p "cmake/build"
+pushd "cmake/build"
+cmake \
+  -DCMAKE_BUILD_TYPE=Release \
+  -DgRPC_INSTALL=ON \
+  -DgRPC_BUILD_TESTS=OFF \
+  -DgRPC_SSL_PROVIDER=package \
+  ../..
+make -j4 protoc grpc_cpp_plugin
+popd
+
+# Download raspberry pi toolchain.
+mkdir -p "/tmp/raspberrypi_root"
+pushd "/tmp/raspberrypi_root"
+git clone https://github.com/raspberrypi/tools raspberrypi-tools
+cd raspberrypi-tools && git checkout 4a335520900ce55e251ac4f420f52bf0b2ab6b1f && cd ..
+
+# Write a toolchain file to use for cross-compiling.
+cat > toolchain.cmake <<'EOT' 
+SET(CMAKE_SYSTEM_NAME Linux)
+SET(CMAKE_SYSTEM_PROCESSOR arm)
+set(devel_root /tmp/raspberrypi_root)
+set(CMAKE_STAGING_PREFIX ${devel_root}/stage)
+set(tool_root ${devel_root}/raspberrypi-tools/arm-bcm2708)
+set(CMAKE_SYSROOT ${tool_root}/arm-rpi-4.9.3-linux-gnueabihf/arm-linux-gnueabihf/sysroot)
+set(CMAKE_C_COMPILER ${tool_root}/arm-linux-gnueabihf/bin/arm-linux-gnueabihf-gcc)
+set(CMAKE_CXX_COMPILER ${tool_root}/arm-linux-gnueabihf/bin/arm-linux-gnueabihf-g++)
+set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
+set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
+set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
+set(CMAKE_FIND_ROOT_PATH_MODE_PACKAGE ONLY)
+EOT
+popd
+
+# Build and install gRPC for raspberry pi.
+mkdir -p "cmake/raspberrypi_build"
+pushd "cmake/raspberrypi_build"
+cmake -DCMAKE_TOOLCHAIN_FILE=/tmp/raspberrypi_root/toolchain.cmake \
+      -DCMAKE_BUILD_TYPE=Release \
+      -DCMAKE_INSTALL_PREFIX=/tmp/raspberrypi_root/grpc_install \
+      -D_gRPC_PROTOBUF_PROTOC_EXECUTABLE="$grpc_dir/cmake/build/third_party/protobuf/protoc" \
+      -D_gRPC_CPP_PLUGIN="$grpc_dir/cmake/build/grpc_cpp_plugin" \
+      ../..
+make -j4 install
+popd
+
+# Build helloworld example for raspberry pi.
+mkdir -p "examples/cpp/helloworld/cmake/raspberrypi_build"
+pushd "examples/cpp/helloworld/cmake/raspberrypi_build"
+cmake -DCMAKE_TOOLCHAIN_FILE=/tmp/raspberrypi_root/toolchain.cmake \
+      -DCMAKE_BUILD_TYPE=Release \
+      -DProtobuf_DIR=/tmp/raspberrypi_root/stage/lib/cmake/protobuf \
+      -D_PROTOBUF_PROTOC="$grpc_dir/cmake/build/third_party/protobuf/protoc" \
```

Comment:
as said above, let's not set `_PROTOBUF_PROTOC` and `_GRPC_CPP_PLUGIN_EXECUTABLE`, but rely on them being on PATH?